db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -1304,6 +1304,14 @@ class NonBatchedOpsStressTest : public StressTest {
         (value_base % FLAGS_use_put_entity_one_in) == 0) {
       s = db_->PutEntity(write_opts, cfh, k,
                          GenerateWideColumns(value_base, v));
+    } else if (FLAGS_use_timed_put_one_in > 0 &&
+               (value_base % FLAGS_use_timed_put_one_in) == 0) {
```

Comment:
If use_put_entity_one_in and use_timed_put_one_in have common factors (or are equal) then then the actual frequency of TimedPut is really messed up or zero. A partial fix would add a reasonably large prime constant to `value_base` before this mod operation.

cache_bench_tool.cc (https://github.com/facebook/rocksdb/blob/main/cache/cache_bench_tool.cc) has an example of efficiently, randomly choosing disjoint actions with chosen frequencies. Search for `lookup_insert_threshold`. (Modulo operator is slow btw.)

Reply:
That's a great point! I was only thinking that `used_timed_put_one_in` will be inaccurate if put entity is also enabled, but didn't realize in the case when they have a common factor, the former will be as if it's unset. Thanks for calling that out and the example for how to solve it. Let me add the partial fix of adding a reasonable large prime. And I will convert these `one_in` type of arguments to be percentage based and use the strategy in cache bench tool for them in a follow up.

Reply:
one_in is the standard for db_stress, so that's fine. I would pointing to the cache_bench code for an example of radomized behavior with exclusive probabilities.